docs: fix linter errors for ajs-quick-reference page.

regulas1 opened the pull request:
This addresses a part of the issue
 https://github.com/angular/angular/issues/47179

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

dario-piotrowicz committed:
fix(compiler): scope css keyframes in emulated view encapsulation (#42608)

Ensure that keyframes rules, defined within components with emulated
view encapsulation, are scoped to avoid collisions with keyframes in
other components.

This is achieved by renaming these keyframes to add a prefix that makes
them unique across the application.

In order to enable the handling of keyframes names defined as strings
the previous strategy of replacing quoted css content with `%QUOTED%`
(introduced in commit 7f689a2) has been removed and in its place now
only specific characters inside quotes are being replaced with
placeholder text (those are `;`, `:` and `,`, more can be added in
the future if the need arises).

Closes #33885

BREAKING CHANGE:

Keyframes names are now prefixed with the component's "scope name".
For example, the following keyframes rule in a component definition,
whose "scope name" is host-my-cmp:

   @keyframes foo { ... }

will become:

   @keyframes host-my-cmp_foo { ... }

Any TypeScript/JavaScript code which relied on the names of keyframes rules
will no longer match.

The recommended solutions in this case are to either:
- change the component's view encapsulation to the `None` or `ShadowDom`
- define keyframes rules in global stylesheets (e.g styles.css)
- define keyframes rules programmatically in code.

PR Close #42608

JoostK committed:
fix(compiler-cli): exclude abstract classes from `strictInjectionParameters` requirement (#44615)

In AOT compilations, the `strictInjectionParameters` compiler option can
be enabled to report errors when an `@Injectable` annotated class has a
constructor with parameters that do not provide an injection token, e.g.
only a primitive type or interface.

Since Ivy it's become required that any class with Angular behavior
(e.g. the `ngOnDestroy` lifecycle hook) is decorated using an Angular
decorator, which meant that `@Injectable()` may need to have been added
to abstract base classes. Doing so would then report an error if
`strictInjectionParameters` is enabled, if the abstract class has an
incompatible constructor for DI purposes. This may be fine though, as
a subclass may call the constructor explicitly without relying on
Angular's DI mechanism.

Therefore, this commit excludes abstract classes from the
`strictInjectionParameters` check. This avoids an error from being
reported at compile time. If the constructor ends up being used by
Angular's DI system at runtime, then the factory function of the
abstract class will throw an error by means of the `ɵɵinvalidFactory`
instruction.

In addition to the runtime error, this commit also analyzes the inheritance
chain of an injectable without a constructor to verify that their inherited
constructor is valid.

BREAKING CHANGE: Invalid constructors for DI may now report compilation errors

When a class inherits its constructor from a base class, the compiler may now
report an error when that constructor cannot be used for DI purposes. This may
either be because the base class is missing an Angular decorator such as
`@Injectable()` or `@Directive()`, or because the constructor contains parameters
which do not have an associated token (such as primitive types like `string`).
These situations used to behave unexpectedly at runtime, where the class may be
constructed without any of its constructor parameters, so this is now reported
as an error during compilation.

Any new errors that may be reported because of this change can be resolved either
by decorating the base class from which the constructor is inherited, or by adding
an explicit constructor to the class for which the error is reported.

Closes #37914

PR Close #44615

arturovt committed:
fix(zone.js): cancel tasks only when they are scheduled or running (#46435)

Currently, there's no check if the task (that is being canceled) has the right state.
Only `scheduled` and `running` tasks can be canceled. If the task has a non-appropriate
state, then an error will be thrown. Cancelation should not throw an error on an already
canceled task, e.g. `clearTimeout` does not throw errors when it's called multiple times
on the same timer.

PR Close #45711

PR Close #46435

atscott committed:
feat(router): make RouterOutlet name an Input so it can be set dynamically (#46569)

This commit updates the 'name' of `RouterOutlet` to be an `Input` rather
than an attribute. Note that this change does not affect `[attr.name]=`
because those already would not have worked. The static name was only
read in the constructor and if it wasn't available, it would use
'PRIMARY' instead.

fixes #12522

BREAKING CHANGE: Previously, the `RouterOutlet` would immediately
instantiate the component being activated during navigation. Now the
component is not instantiated until the change detection runs. This
could affect tests which do not trigger change detection after a router
navigation. In rarer cases, this can affect production code that relies
on the exact timing of component availability.

PR Close #46569

matthiasweiss committed:
feat(common): add injection token for default DatePipe configuration (#47157)

This commit introduces a new `DATE_PIPE_DEFAULT_OPTIONS` token, which
can be used to configure default DatePipe options, such as date
format and timezone.

DEPRECATED:

The `DATE_PIPE_DEFAULT_TIMEZONE` token is now deprecated in favor
of the `DATE_PIPE_DEFAULT_OPTIONS` token, which accepts an object
as a value and the timezone can be defined as a field (called `timezone`)
on that object.

PR Close #47157

Commit:
docs: update of documentation contrib. guide (#47381)

This update describes the content authoring and revision process in sufficient detail for technical and non-technical contributors to the angular.io documentation.

PR Close #47381

abergquist committed:
docs: Separated HTML elements with blank lines (#47331)

Separated HTML elements with judicious use of space. It not only makes it easier to read the HTML; it also makes it consistent with the formatting of https://github.com/angular/angular/edit/main/aio/content/examples/what-is-angular/src/app/hello-world-bindings/hello-world-ngif.component.html.

PR Close #47331

yharaskrik committed:
feat(common): add <link> preload tag on server for priority img (#47343)

This commit adds a logic that generates preload tags for priority images, when rendering happens on the server (e.g. Angular Universal).

PR Close #47343

DcsPeterDickten committed:
docs: fixed typo in AsyncValidatorFn (#47465)

the return type must be
Observable<ValidationErrors | null>
instead of
Observable<ValidationErrors> | null

PR Close #47465

devversion committed:
build: use shared circleci dev-infra for common circleci code (#47467)

This allows us to remove the duplicated `rebase-pr.js` script so that we can maintain this script in a single place.

PR Close #47467

alxhub committed:
refactor(http): introduce the functional interceptors API (#47502)

This commit introduces new types and symbols related to functional HTTP
interceptors - interceptors which are plain functions with access to DI via
the `inject()` operation.

This new form of interceptor is not exposed publically in this commit, but
the legacy class-based interceptors are refactored to be built on top of the
new API internally.

PR Close #47502

alxhub committed:
refactor(http): use a DI token for new interceptors (#47502)

This commit introduces a new DI token for the set of functional
interceptors. This is a no-op in terms of behavior currently, but will allow
for the deduplication of the bridge interceptor which connects legacy class-
based interceptors to the functional interceptor chain.

PR Close #47502

alxhub committed:
refactor(http): convert the JSONP interceptor to the functional style (#47502)

This commit rewrites the JSONP interceptor to use the functional interceptor
style internally, while still maintaining the same public API and behavior.

PR Close #47502

alxhub committed:
refactor(http): convert the XSRF interceptor to functional style (#47502)

This commit converts the XSRF interceptor into a functional interceptor
instead of a legacy class-based interceptor.

PR Close #47502

alxhub committed:
feat(http): introduce the `provideHttpClient()` API (#47502)

This commit introduces the main components of the `provideHttpClient()`
provider API, designed in the style of `provideRouter()`. Initial features
are defined for including legacy class-based interceptors, JSONP support,
and configuring or disabling the builtin XSRF protection.

This API is an alternative to providing `HttpClient` via the
`HttpClientModule`, and is more tree-shakable and more capable than the
NgModule implementation.

Tests are included to validate the new configuration format as well as the
interoperability of the two styles of providing and configuring
`HttpClient`.

PR Close #47502

alxhub committed:
feat(http): introduce functional interceptors (#47502)

This commit introduces a new feature for `provideHttpClient` called
`withInterceptors`. This feature exposes and configures the new concept of
functional interceptors.

Functional interceptors use functions instead of classes to implement an
HTTP interceptor. Such interceptor functions have access to the DI context
from the `EnvironmentInjector` in which they're configured via the
`inject()` function. Otherwise, functional interceptors are identical in
capability to the existing interceptor system.

PR Close #47502

alxhub committed:
feat(http): allow for child `HttpClient`s to request via parents (#47502)

Ordinarily, providing `HttpClient` (either via `provideHttpClient` or the
`HttpClientModule`) creates an entirely separate HTTP context. Requests made
via that client are not passed through the interceptor chains that are
configured in a parent injector, for this example.

This commit introduces a new option for `provideHttpClient` called
`withRequestsMadeViaParent()`. When this option is passed, requests made in
the child context flow through any injectors, etc. and are then handed off
to the parent context.

This addresses a longstanding issue with interceptors where it's not
possible to extend the set of interceptors in a child context without
repeating all of the interceptors from the parent.

PR Close #47502

alxhub committed:
docs(http): fix copy-paste error in HttpClientJsonpModule docs (#47502)

This commit deletes a sentence from the `HttpClientJsonpModule` docs which
was accidentally copy-pasted from the docs for another symbol.

PR Close #47502

alxhub committed:
refactor(http): use an InjectionToken to disable the XSRF interceptor (#47502)

@angular/common/http has XSRF protection which is enabled by default and is
implemented as an interceptor. Previously, this protection could be disabled
with an API which would internally provide a `NoopInterceptor` in place of
the standard XSRF interceptor.

To achieve the same capability of disabling the XSRF interceptor after it is
converted to the functional style, an InjectionToken is added in this commit
which disables the XSRF interceptor. This way, the interceptor can be
disabled in place without needing to override it via DI (which is difficult
for functional interceptors).

PR Close #47502

alxhub committed:
feat(http): introduce `provideHttpClientTesting` provider function (#47502)

This commit introduces the `provideHttpClientTesting()` function as an
alternative to the `HttpClientTestingModule` (in fact, the NgModule is
converted internally to just use the new provider function).

PR Close #47502

alxhub committed:
refactor(http): provide XSRF configuration tokens in root (#47502)

This commit switches the XSRF configuration tokens (for header name and
cookie name) to be `providedIn: 'root'`. This is a no-op change now as they
are always provided along with any usage of them via `HttpClientModule`, but
will become load-bearing as the `provideHttpClient` API will not provide
these tokens, and will rely on injecting them from either the parent context
or from these root providers.

PR Close #47502

alxhub committed:
refactor(http): convert NgModules to use `provideHttpClient` internally (#47502)

This commit converts `HttpClientModule` to use `provideHttpClient()`
internally, with a particular configuration of features. Other NgModules
related to configuring `HttpClient` are also converted to use the providers
directly from various features, to ensure consistency of behavior.

PR Close #47502

angular-robot committed:
build: update io_bazel_rules_sass digest to f6ceac7 (#47376)

See associated pull request for more information.

PR Close #47376

MathewBerg committed:
docs: fix documentation linting errors in ajs-quick-reference.md (#47526)

PR Close #47526

alexanderwiebe committed:
fix(docs-infra): add punctuation to dr iq (#47525)

add punctuation to dr iq name to aid in clarity

Fixes #46011

PR Close #47525

devversion committed:
build: setup remote execution through shared command (#47537)

Sets up remote execution through a shared command.

PR Close #47537

alvaromartmart committed:
docs(forms): setErrors emitEvent default value (#47546)
PR Close #47546

cipriansauliuc committed:
docs: add oxford commas and rephrase sentences (#47540)

PR Close #47540

crisbeto committed:
refactor(core): add input and output filtering for host directives (#47536)

Adds the logic that will filter out unexposed inputs/outputs and apply the aliases that the author specified when writing the host directives.

PR Close #47536

pkozlowski-opensource committed:
refactor(common): decouple preconnect checks from image loaders (#47565)

Previously the built-in image loaders for the optimized image directive
were tightly coupled to the preconnect checks infrastructure. This was
creating a problem when developers were trying to provide a loader on
a component level (or, more generally, deeper in the DI hierarchy):
- PreconnectLinkChecker is the application-level service, provided in root;
- it makes sense to provide loaders in different parts of the DI hierarchy;

This refactoring removes the PreconnectLinkChecker configuration from
the loaders infrastructure and makes it application-wide config. The
PRECONNECT_CHECK_BLOCKLIST is also a simple provider (it was a
multi-provider previously) which should make the overall configuration
easier.

PR Close #47565

pkozlowski-opensource committed:
test(common): demonstrate that image loaders are configurable on a component level (#47565)

Add a test to ensure that optimized image loaders can be configured at any
level of the DI hierarchy, including component's node injector.

PR Close #47565

angular-robot committed:
build: update cross-repo angular dependencies (#47454)

See associated pull request for more information.

PR Close #47454

alan-agius4 committed:
fix(localize): add polyfill in polyfills array instead of polyfills.ts (#47569)

With the recent changes in the Angular CLI (https://github.com/angular/angular-cli/pull/23938) the polyfills option accepts module path that are resolved using Node module resolution. Also, the polyfills.ts file is no longer generated by default.

This commit changes the way on how the polyfill is added to the projects.

PR Close #47569

alan-agius4 committed:
build: update Angular tooling packages to `15.0.0-next.3` (#47584)

- This update is needed to implement the changes in `ng add localize` https://github.com/angular/angular/pull/47569
- Add missing `root` options to all `angular.json`, this is required as otherwise the angular.json validation will fail.
- Remove `require.context` from test.ts integration test, as this is no longer needed.
- Update payloads golden files.

PR Close #47584

alan-agius4 committed:
build: update puppeteer to 18.0.5 (#47584)

This is needed as Angular does not support Chrome 93.

PR Close #47584

mprobst committed:
fix(compiler-cli): support hasInvalidatedResolutions. (#47585)

The latest TypeScript compiler exposes the previously private field
`hasInvalidatedResolutions`. That breaks Angular in the newer TS,
because the new field would be required on DelegatingCompilerHost.
However we cannot just add the field here, because it's not present in
the older compiler.

This change adds the field for delegation, which works at runtime
because the field is present. It suppresses the compiler error using a
`// @ts-expect-error`, which should be removed once Angular moves to a
TSC version that includes this change.

PR Close #47585

alan-agius4 committed:
test: delete bazel_ngtsc_plugin integration test (#47584)

This does not provide any added value.

PR Close #47584

alxhub committed:
feat(router): auto-unwrap default exports when lazy loading (#47586)

When using `loadChildren` or `loadComponent`, a common pattern is to pass
a function that returns a `Promise` from a dynamic import:

```typescript
{
  path: 'lazy',
  loadComponent: () => import('./lazy-file').then(m => m.LazyCmp),
}
```

The `.then` part of the expression selects the particular exported
component symbol from the dynamically imported ES module.

ES modules can have a "default export", created with the `export default`
modifier:

```typescript
@Component({...})
export default class LazyCmp { ... }
```

This default export is made available to dynamic imports under the well-
known key of `'default'`, per the ES module spec:
https://tc39.es/ecma262/#table-export-forms-mapping-to-exportentry-records

This commit adds a feature to the router to automatically dereference such
default exports. With this logic, when `export default` is used, a `.then`
operation to select the particular exported symbol is no longer required:

```typescript
{
  path: 'lazy',
  loadComponent: () => import('./lazy-file'),
}
```

The above `loadComponent` operation will automatically use the `default`
export of the `lazy-file` ES module.

This functionality works for `loadChildren` as well.

PR Close #47586

alxhub committed:
docs(core): deprecate `providedIn: NgModule` and `providedIn: 'any'` (#47616)

This commit deprecates the above two forms of `providedIn`.

DEPRECATED:

- The ability to pass an `NgModule` to the `providedIn` option for
`@Injectable` and `InjectionToken` is now deprecated.

`providedIn: NgModule` was intended to be a tree-shakable alternative to
NgModule providers. It does not have wide usage, and in most cases is used
incorrectly, in circumstances where `providedIn: 'root'` should be
preferred. If providers should truly be scoped to a specific NgModule, use
`NgModule.providers` instead.

- The ability to set `providedIn: 'any'` for an `@Injectable` or
`InjectionToken` is now deprecated.

`providedIn: 'any'` is an option with confusing semantics and is almost
never used apart from a handful of esoteric cases internal to the framework.

PR Close #47616

crisbeto committed:
refactor(core): add host directive definitions validation (#47589)

Adds some logic to ensure that host directives are configured correctly.

PR Close #47589

gkalpak committed:
build(docs-infra): upgrade cli command docs sources to 77c70052e (#47590)

Updating [angular#main](https://github.com/angular/angular/tree/main) from
[cli-builds#main](https://github.com/angular/cli-builds/tree/main).

##
Relevant changes in
[commit range](https://github.com/angular/cli-builds/compare/556a775f3...77c70052e):

**Modified**
- help/completion.json

PR Close #47590

markostanimirovic committed:
docs: add Marko to GDE resources (#47595)

PR Close #47595

crisbeto committed:
refactor(core): handle ngOnChanges in host directives (#47597)

The `NgOnChanges` feature matches some legacy ViewEngine behavior where the keys in the `SimpleChanges` object are based on the *declared* names of the inputs, not the public or minified names. This is achieved by constructing the `DirectiveDef.declaredInputs` object at the same time as when `DirectiveDef.inputs` is created.

This logic breaks down for host directives, because they can re-alias the input under a different public name which won't be present in the `declaredInputs`.

These changes add some logic to patch the directive def aliases onto the `declaredInputs`. There is some validation in place to ensure that this patching doesn't overwrite any pre-existing inputs.

PR Close #47597

jordanpowell88 committed:
docs: add Jordan Powell to GDE resources (#47598)

PR Close #47598

AndrewKushnir committed:
feat(router): migrate `RouterLinkWithHref` references to `RouterLink` (#47599)

Since Angular v15, the `RouterLink` contains the logic of the `RouterLinkWithHref` directive and now developers can always import and use the `RouterLink` directive when they need to add a `[routerLink]` in templates. This migration finds all imports and usages of the `RouterLinkWithHref` class and rewrites them to `RouterLink` instead.

```ts
import { RouterLinkWithHref } from '@angular/router';

@Component({
  standalone: true,
  template: `<a [routerLink]="'/abc'">`,
  imports: [RouterLinkWithHref]
})
export class MyComponent {
  @ViewChild(RouterLinkWithHref) aLink!: RouterLinkWithHref;
}
```

```ts
import { RouterLink } from '@angular/router';

@Component({
  standalone: true,
  template: `<a [routerLink]="'/abc'">`,
  imports: [RouterLink]
})
export class MyComponent {
  @ViewChild(RouterLink) aLink!: RouterLink;
}
```

PR Close #47599

DMezhenskyi committed:
docs: add missing link to canMatch guard (#47602)

PR Close #47602

AndrewKushnir committed:
feat(router): add a migration to remove `relativeLinkResolution` usages (#47604)

As of Angular v15, the deprecated `relativeLinkResolution` config option of the Router is removed.  This migration cleans up (removes) the `relativeLinkResolution` fields from the Router config objects in applications code.

```ts
import { RouterModule } from '@angular/router';

RouterModule.forRoot([], {
  relativeLinkResolution: 'legacy',
  enableTracing: false,
});
```

```ts
import { RouterModule } from '@angular/router';

RouterModule.forRoot([], {
  // the `relativeLinkResolution` is removed
  enableTracing: false,
});
```

PR Close #47604

AndrewKushnir committed:
refactor(router): apply `href` value via renderer to further align RouterLink directives (#47605)

This commit updates the `RouterLink` directive to use a renderer to apply an `href` attribute value. Previously, the `RouterLinkWithHref` used a host binding for this and that prevented further merge of the `RouterLink` and `RouterLinkWithHref` directives.

PR Close #47605

crisbeto committed:
fix(compiler): update element schema (#47552)

Updates the DOM element schema to add the `hgroup` element and to include some properties that weren't supported the last time the schema was updated.

Fixes #47545.

PR Close #47552

AndrewKushnir committed:
ci: cleanup PullApprove config file (#47611)

This commit updates the PullApprove config file.

PR Close #47611

alxhub committed:
refactor(router): organize deprecated loadChildren code (#47586)

This commit breaks out the code in @angular/router needed to support the
deprecated `loadChildren` signature into a separate file, which reduces the
opportunity for conflicts when patching that behavior in g3.

PR Close #47586

dsnoeijer committed:
docs: fix linter errors for angular-compiler-options.md page (#47179) (#47618)

fixup! docs: fix linter errors for angular-compiler-options.md page (#47179)

PR Close #47618

AndrewKushnir committed:
refactor(router): drop own constructor from the `RouterLinkWithHref` class (#47619)

This commit updates the `RouterLinkWithHref` class to further align with the `RouterLink` class by removing own constructor from the `RouterLinkWithHref` class.

PR Close #47619

dylhunn committed:
refactor(compiler): Rename `DirectiveInScope` -> `PotentialDirective` (#47561)

After implementing `getPotentialTemplateDirectives`, we will use this data struture to represent both in-scope and out-of-scope directives. So this rename is an advance cleanup.

PR Close #47561

crisbeto committed:
refactor(core): support host directives on a root component (#47620)

Adds support for host directives on the root component by calling `findHostDirectiveDefs` and passing the results to the relevant functions.

PR Close #47620

atscott committed:
fix(router): Remove deprecated relativeLinkResolution (#47623)

The `relativeLinkResolution` is no longer supported. The only behavior
now is the correct behavior.

BREAKING CHANGE: `relativeLinkResolution` is no longer configurable in
the Router. This option was used as a means to opt out of a bug fix.

PR Close #47623

crisbeto committed:
feat(core): enable the new directive composition API (#47642)

Enables the new directive composition API by exposing the `hostDirectives` property on the `Directive` and `Component` decorators. Also cleans up some casts that were put in place while the feature was being developed.

Fixes #8785.

PR Close #47642

dylhunn committed:
refactor(compiler): Add `getPotentialTemplateDirectives` API method. (#47561)

`getPotentialTemplateDirectives` returns possible directives which can be used in the provided context, whether already in scope or requiring an import.

This is necessary to implement auto-import support for standalone components in the language service.

PR Close #47561

dylhunn committed:
refactor(compiler): Rename `PipeInScope` -> `PotentialPipe` (#47561)

Now that `getPotentialTemplateDirectives` uses `PotentialDirective`, we should rename `PipeInScope` to match.

PR Close #47561

atscott committed:
test(router): Ensure change detection runs after navigation (#47629)

Components are not fully rendered until change detection runs.

PR Close #47629

AndrewKushnir committed:
feat(router): merge `RouterLinkWithHref` into `RouterLink` (#47630)

This commit updates the `RouterLink` to extend the selector to also include `<a>` and `<area>` elements, which were previously matched by the `RouterLinkWithHref` directive. The code of the directives was merged together (since there was a lot of duplication) and this commit finalizes the merge. The `RouterLinkWithHref` becomes an alias of the `RouterLink` directive.

Now there is no need to import and use the `RouterLinkWithHref` class, the `RouterLink` directive will match any element that has the `routerLink` attribute.

DEPRECATED:

The `RouterLinkWithHref` directive is deprecated, use the `RouterLink` directive instead. The `RouterLink` contains the code from the `RouterLinkWithHref` to handle elements with `href` attributes.

PR Close #47630

dylhunn committed:
refactor(compiler): Add `getPotentialImportsFor` method on template type checker (#47631)

`getPotentialImportsFor` returns an array of possible imports, including TypeScript module specifier and identifier name, for a requested trait in the context of a given component.

PR Close #47631

dylhunn committed:
feat(language-service): Quick fix to import a component when its selector is used (#47088)

The language service can now generate an import corresponding to a selector. This includes both the TypeScript module import and the decorator import. This applies to both standalone components and components declared in NgModules.

PR Close #47088

regulas1 committed:
docs: fix linter errors for ajs-quick-reference page.

This addresses a part of the issue
 https://github.com/angular/angular/issues/47179

regulas1 committed:
docs: fix linter errors for ajs-quick-reference page.

This addresses a part of the issue
 https://github.com/angular/angular/issues/47179

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

mprobst committed:
fix(compiler-cli): use @ts-ignore. (#47636)

The previous commit 2e1dddec45 used `@ts-expect-error` to suppress the
current error, with the intent of being informed once that's no longer
an error, ie. when we updated to an upstream TS version that includes
this change.

However this unfortunately means the change is incompatible with the
fixed version, which prevents it from working with an updated TS version
in google3.

This change reverts back to the original `@ts-ignore` which is forwards
and backwards compatible, avoiding that problem (but unfortunately
losing the benefit of being notified once fixed).

PR Close #47636

AndrewKushnir committed:
docs: release notes for the v14.2.5 release

AndrewKushnir committed:
release: cut the v15.0.0-next.5 release

dsnoeijer committed:
docs: fix linter errors for lazy-loading-ngmodules.md (#47645)

Update aio/content/guide/lazy-loading-ngmodules.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

PR Close #47645

jessicajaniuk commented:
@regulas1 This change looks good, but looks like you'll need to rebase. CI is failing to run due to being behind.

clydin committed:
perf(compiler-cli): minimize filesystem calls when generating shims (#47682)

Previously when a file was being analyzed to determine if a shim should
be generated, up to two calls to the host `fileExists` function per file
per generator were made. In the default host, each `fileExists` call made
two underlying file system calls. Following these calls, the file was then
read via `getSourceFile`. However, `getSourceFile` will return `undefined`
if the requested file does not exist. As a result, `getSourceFile` can be
used directly to request both potential file names and leverage the return
value to determine if the file does not exist. This avoids the need to call
`fileExists` at all.

PR Close #47682

alxhub committed:
refactor(core): define shape of EnvironmentProviders interface (#47669)

This commit introduces the `EnvironmentProviders` interface, but does not
yet export it as public API.

PR Close #47669

alxhub committed:
refactor(core): support `EnvironmentProviders` types internally (#47669)

This commit modifies `R3Injector` and other code in Angular that deals with
providers, to handle `EnvironmentProviders` objects as well as normal
`Provider` types. There is no user-visible impact to this change, but it
prepares the core of the DI system for the introduction of
`EnvironmentProviders` as a public feature.

PR Close #47669

alxhub committed:
feat(core): introduce EnvironmentProviders wrapper type (#47669)

This commit introduces a new type `EnvironmentProviders` which can be used
in contexts where Angular accepted `Provider`s destined for
`EnvironmentInjector`s. This includes contexts such as `@NgModule.providers`
and `Route.providers`.

The new type is useful for preventing such providers from accidentally
ending up in `@Component.providers`. It can be used as the return type of
provider functions (such as `provideRouter`) to enforce this safety.

Because `Provider` allows `any[]` nested arrays, the compile-time safety
provided by `EnvironmentProviders` is easily circumvented. However, the
runtime shape of `EnvironmentProviders` is not compatible with component
injectors and will result in a runtime error if it leaks through (NG0207).

A new function `makeEnvironmentProviders` is used to construct this new type
from an array of providers.

The existing `importProvidersFrom` operation previously returned a very
similar type `ImportedNgModuleProviders` which had the same goal. This
machinery is switched over to use the new `EnvironmentProviders` interface
instead (in fact, `ImportedNgModuleProviders` is now just an alias to
`EnvironmentProviders`).

PR Close #47669

atscott committed:
build: add atscott to unavailable list for pullapprove (#47658)

atscott is unavailable to review PRs until no longer OOO

PR Close #47658

alxhub committed:
feat(router): prevent `provideRouter()` from usage in @Component (#47669)

This commit switches `provideRouter()` to return the new
`EnvironmentProviders` wrapper type, preventing it from being accidentally
(or intentionally) included in `@Component.providers`.

PR Close #47669

dsnoeijer committed:
docs: fix linter errors for libraries.md (#47668)

PR Close #47668

dsnoeijer committed:
docs: fix linter errors for lightweight-injection-tokens.md (#47673)

PR Close #47673

AndrewKushnir committed:
feat(common): add `provideLocationMocks()` function to provide Location mocks (#47674)

This commit adds the `provideLocationMocks()` function that returns mocks for the `Location` and `LocationStrategy` classes. This function can be used in tests to configure an environment where it's possible to fire simulated location events (helpful when testing Router configuration).

PR Close #47674

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -56,7 +56,7 @@ The following are some of the key AngularJS built-in directives and their equiva
 | <header><code>ng-click</code></header> <code-example hideCopy format="html" language="html"> &lt;button ng-click="vm.toggleImage()"&gt; &NewLine; &lt;button ng-click="vm.toggleImage(&dollar;event)"&gt; </code-example> In AngularJS, the `ng-click` directive allows you to specify custom behavior when an element is clicked. <br /> In the first example, when the user clicks the button, the `toggleI ...
 
-### `ng-controller` &rarr; Component decorator
+### `ng-controller` &rarr; Component Decorator
```
This is a tricky one. `Component` is a code reference, so it should be marked as code to indicate such and remove the lint error.
```suggestion
### `ng-controller` &rarr; `Component` decorator
```

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -84,7 +84,7 @@ The following are some of the key AngularJS built-in directives and their equiva
 | AngularJS                                                                                                                                                                                                                                                                                                                                                                                                     ...
 |:---                                                                                                                                                                                                                                                                                                                                                                                                           ...
-| <header><code>ng-model</code></header> <code-example hideCopy format="html" language="html"> &lt;input ng-model="vm.favoriteHero" /&gt; </code-example> In AngularJS, the `ng-model` directive binds a form control to a property in the controller associated with the template. This provides **two-way binding**, whereby any change made to the value in the view is synchronized with the model, and any ...
+| <header><code>ng-model</code></header> <code-example hideCopy format="html" language="html"> &lt;input ng-model="vm.favoriteHero" /&gt; </code-example> In AngularJS, the `ng-model` directive binds a form control to a property in the controller associated with the template. This provides **two-way binding**, whereby any change made to the value in the view is synchronized with the model, and any ...
```
This has some difficult text:
 ```suggestion
| <header><code>ng-model</code></header> <code-example hideCopy format="html" language="html"> &lt;input ng-model="vm.favoriteHero" /&gt; </code-example> In AngularJS, the `ng-model` directive binds a form control to a property in the controller associated with the template. This provides **two-way binding**, whereby any change made to the value in the view is synchronized with the model, and any change to the model is synchronized with the value in the view. | <header><code>ngModel</code></header> <code-example hideCopy path="ajs-quick-reference/src/app/movie-list.component.html" region="ngModel"></code-example> In Angular, **two-way binding** is indicated by `[()]`, descriptively referred to as a *banana in a box*. This syntax is a shortcut for defining both property binding \(from the component to the view\) and event binding \(from the view to the component\), thereby providing two-way binding. <br /> For more information on two-way binding with `ngModel`, see the [Displaying and updating properties with `ngModel`][AioGuideBuiltInDirectivesDisplayingAndUpdatingPropertiesWithNgmodel] section of [Built-in directives][AioGuideBuiltInDirectives]. |
```
This changes the "banana in a box" reference to *banana in a box* to properly annotate a term and fixes the typo in `indicatedr5t`.

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -112,11 +112,11 @@ The following are some of the key AngularJS built-in directives and their equiva
+|  <header><code>ng-switch</code></header> <code-example hideCopy format="html" language="html"> &lt;div ng-switch="vm.favoriteHero &amp;&amp; vm.checkMovieHero(vm.favoriteHero)"&gt; &NewLine; &nbsp; &lt;div ng-switch-when="true"&gt; &NewLine; &nbsp; &nbsp; Excellent choice. &NewLine; &nbsp; &lt;/div&gt; &NewLine; &nbsp; &lt;div ng-switch-when="false"&gt; &NewLine; &nbsp; &nbsp; No movie, sorry.  ...
 
-## Filters/pipes
+## Filters/**Pipes**
```
```suggestion
## Filters and pipes
```
If you accept this change, be sure to change the corresponding link in line 232 as follows so the docs will build:
```
[AioGuideAjsQuickReferenceFiltersPipes]: guide/ajs-quick-reference#filters-and-pipes "Filters and pipes - AngularJS to Angular concepts: Quick reference | Angular"
```

bob-watson replied:
Changing the heading text changes the anchor to that heading.

Please replace line 232  with the line in the previous comment so that the correct anchor is used by the link.
Also, don't forget to update the `@reviewed` date at the end of the file.

Thanks for your contribution!

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -186,25 +186,25 @@ The Angular code is shown using TypeScript.
 | <header>Angular modules</header> <code-example hideCopy format="typescript" language="typescript"> angular .module( &NewLine;&nbsp; "movieHunter", &NewLine;&nbsp; [ &NewLine;&nbsp;&nbsp;&nbsp; "ngRoute" &NewLine;&nbsp; ] &NewLine;); </code-example> In AngularJS, an Angular module keeps track of controllers, services, and other code. The second argument defines the list of other modules that thi ...
 
-### Controller registration &rarr; Component decorator
+### Controller Registration &rarr; Component Decorator
```
This is a tricky one as noted in the earlier comment.
```suggestion
### Controller registration &rarr; `Component` decorator
```

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -186,25 +186,25 @@ The Angular code is shown using TypeScript.
 | <header>Controller registration</header> <code-example hideCopy format="typescript" language="typescript"> angular .module( &NewLine;&nbsp; "movieHunter" &NewLine;) .controller( &NewLine;&nbsp; "MovieListCtrl", &NewLine;&nbsp; [ &NewLine;&nbsp;&nbsp;&nbsp; "movieService", &NewLine;&nbsp;&nbsp;&nbsp; MovieListCtrl &NewLine;&nbsp; ] &NewLine;); </code-example> AngularJS has code in each controlle ...
 
-### Controller function &rarr; Component class
+### Controller Function &rarr; Component Class
```
Same as previous
```suggestion
### Controller function &rarr; `Component` class
```

bob-watson commented on aio/content/guide/ajs-quick-reference.md:
```diff
@@ -186,25 +186,25 @@ The Angular code is shown using TypeScript.
+| <header>Controller function</header> <code-example hideCopy format="typescript" language="typescript"> function MovieListCtrl(movieService) { &NewLine; } </code-example> In AngularJS, you write the code for the model and methods in a controller function. | <header>Component class</header> <code-example hideCopy path="ajs-quick-reference/src/app/movie-list.component.ts" region="class"></code-exa ...
 
-### Dependency injection &rarr; Dependency injection
+### Dependency Injection &rarr; Dependency Injection
```
```suggestion
### Dependency injection &rarr; dependency injection
```

bob-watson reviewed (changes requested):
Thanks for tackling this--especially for your first PR!
This is a particularly challenging topic because it contains terms from AngularJS as well as Angular. As such, trying to make the linter happy is difficult. Nevertheless, you were able to get rid of quite a few errors!

I left a few comments and suggestions below.

Also, don't forget to update the @reviewed date!

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Update aio/content/guide/ajs-quick-reference.md

Co-authored-by: Bob Watson <104218420+bob-watson@users.noreply.github.com>

regulas1 committed:
Merge branch 'angular:main' into DocumentationUpdate47179

angular-robot committed:
build: update cross-repo angular dependencies (#47600)

See associated pull request for more information.

PR Close #47600

bob-watson commented:
Thanks for the update!
Please see the comment in [the reopened comment](https://github.com/angular/angular/pull/47635#discussion_r988475281). 

Also, after you make that change, could you squash the commits into a single commit? With that, we'll be able to merge this.

Thanks, again for your contribution!

clydin committed:
refactor(compiler-cli): use mkdirSync recursive option instead of custom implementation (#47678)

The supported Node.js versions for the `@angular/compiler-cli` package (^14.15.0 || >=16.10.0)
allow for the use of the `recursive` option of `mkdirSync`.  Using the `recursive` option
removes the need to manually create each subdirectory in a given path.

PR Close #47678

Commit:
refactor(docs-infra): add new Input for better description of close button in notification component (#47681)

Add a new input for banner description with default value, allowing the insertion of other descriptions and if nothing is passed, it assumes a default value.

PR Close #47681

alxhub committed:
fix(http): better handle unexpected `undefined` XSRF tokens (#47683)

`HttpXsrfTokenExtractor` allows returning `string|null` for an XSRF token,
and the interceptor checked if the returned token is `null`. However, some
implementations return `undefined` instead (behind an `any`) type, which
caused the interceptor to crash when trying to set an `undefined` value for
the header.

This commit makes the XSRF interceptor a little more resilient against such
broken implementations of the `HttpXsrfTokenExtractor` interface.

PR Close #47683

atcastle committed:
feat(common): Add automatic srcset generation to ngOptimizedImage (#47547)

Add a feature to automatically generate the srcset attribute for images using the NgOptimizedImage directive. Uses the 'sizes' attribute to determine the appropriate srcset to generate.

PR Close #47547

angular-robot committed:
build: update scorecard action dependencies (#47385)

See associated pull request for more information.

PR Close #47385

dsnoeijer committed:
docs: fix linter errors for property-binding-best-practices.md (#47686)

PR Close #47686

dsnoeijer committed:
docs: fix linter errors for property-binding.md (#47687)

PR Close #47687

dsnoeijer committed:
docs: fix linter errors for security.md (#47688)

PR Close #47688

regulas1 committed:
Added Filter and pipes to build documentation

regulas1 committed:
docs: Added Filter and pipes to build documentation

regulas1 committed:
Merge branch 'main' into DocumentationUpdate47179

dylhunn committed:
fix(forms): call `setDisabledState` on `ControlValueAcessor` when control is enabled (#47576)

Previously, `setDisabledState` was never called when attached if the control is enabled. This PR fixes the bug, and creates a configuration option to opt-out of the fix.

Fixes #35309.

BREAKING CHANGE: setDisabledState will always be called when a `ControlValueAccessor` is attached. You can opt-out with `FormsModule.withConfig` or `ReactiveFormsModule.withConfig`.

PR Close #47576

crisbeto committed:
feat(core): drop support for TypeScript 4.6 and 4.7 (#47690)

Updates the version range in the compiler to require at least TypeScript 4.8. Note that I'm keeping the backwards-compatibility layer for 4.7 around for now until internal projects have been migrated to 4.8.

BREAKING CHANGE:
TypeScript versions older than 4.8 are no longer supported.

PR Close #47690

Commit:
docs: remove assorted lint errors (#47693)

PR Close #47693

angular-robot committed:
build: update dependency google-closure-compiler to v20221004 (#47696)

See associated pull request for more information.

PR Close #47696

jessicajaniuk commented:
@regulas1 Looks like you've merged main into your branch instead of rebasing and squashing. It seems to be causing issues with CI and conflicts. Can you fix that?

Commit:
docs: update spelling dictionary (#47697)

PR Close #47697

regulas1 commented:
> @regulas1 Looks like you've merged main into your branch instead of rebasing and squashing. It seems to be causing issues with CI and conflicts. Can you fix that?

Yes I will fix that. Apologies.

sriram-km committed:
docs(core): Improved README.md (#47700)

PR Close #47700

angular-robot committed:
build: update dependency @rollup/plugin-babel to v6 (#47701)

See associated pull request for more information.

PR Close #47701

gkalpak committed:
build(docs-infra): upgrade cli command docs sources to 46cb32b90 (#47703)

Updating [angular#main](https://github.com/angular/angular/tree/main) from
[cli-builds#main](https://github.com/angular/cli-builds/tree/main).

##
Relevant changes in
[commit range](https://github.com/angular/cli-builds/compare/77c70052e...46cb32b90):

**Modified**
- help/generate.json

PR Close #47703

joshuamorony committed:
docs: add Joshua Morony to GDE resources (#47704)

PR Close #47704

dsnoeijer committed:
docs: fix linter errors for what-is-angular.md (#47708)

PR Close #47708

dsnoeijer committed:
docs: fix linter errors for workspace-config.md (#47709)

PR Close #47709

angular-robot committed:
build: update all non-major dependencies (#47551)

See associated pull request for more information.

PR Close #47551

AleksanderBodurri committed:
build(devtools): migrate to manifest v3 (#47575)

Previously we built DevTools for all browsers with version 2 of the manifest file format.

This commit includes a number of refactors and API additions that will enable us to build DevTools with version 3 of the manifest file format.

The manifest v3 build of Angular DevTools has been tested on Chrome, Edge, and Safari.

Notably, the Firefox version of Angular DevTools remains as a manifest v2 build. Firefox does not yet support manifest v3 in it's latest stable release. When Firefox makes this transition, a follow up PR will update the Firefox manifest file to version 3.

Because Firefox still needs v2, we need to keep some old v2 APIs around in our background page (service worker in v3) that will execute conditionally based on if the extension was built for v2 or v3. This is determined with the chrome.runtime.getManifest().manifest_version API.

PR Close #47575

fmalcher committed:
feat(forms): export forms utility functions: isFormArray, isFormGroup… (#47718)

This commit exports existing utility functions to check for control instances:
isFormControl, isFormGroup, isFormRecord, isFormArray
Those are useful when implementing validators that use the specifics of one of those control types.
To narrow down the type to what it actually is, we can now use the util functions in validators:

```
export const myArrayValidator: ValidatorFn = (control) => {
  if (!isFormArray(control)) { return null; }

  // now you can use FormArray-specific members, e.g.:
  if (control.controls.every(c => !!c.value) {
    return { myerror: true }
  } else { return null; }
}
```

PR Close #47718

angular-robot committed:
build: update dependency @rollup/plugin-commonjs to v23 (#47711)

See associated pull request for more information.

PR Close #47711

josephperrott committed:
ci: adjust workflow permissions for OpenSSF Scorecard results publishing (#47721)

Version 2 of the `openssf/scorecard-action` GitHub action requires the
`id-token: write` permission to publish results.

PR Close #47721

bampakoa committed:
docs: add 0 to filename of runtime error guides (#47722)

PR Close #47722

dsnoeijer committed:
docs: fix linter errors in prerendering.md (#47724)

PR Close #47724

thePunderWoman committed:
Revert "fix(compiler): update element schema" (#47726)

This reverts commit 3c45f6f33e7a20c460d95203a562e0c056d3dbf7.

PR Close #47726

regulas1 committed:
docs: updated ngModel

regulas1 committed:
updated ngModel

regulas1 committed:
Merge branch 'angular:main' into DocumentationUpdate47179

crisbeto committed:
fix(compiler): update element schema (#47552)

Updates the DOM element schema to add the `hgroup` element and to include some properties that weren't supported the last time the schema was updated.

Fixes #47545.

PR Close #47552

alan-agius4 committed:
feat(core): add support for Node.js version 18 (#47730)

This change aligns with the supported Node.js versions of the Angular CLI.
See: https://github.com/angular/angular-cli/pull/24026

BREAKING CHANGE: Angular no longer supports Node.js versions `14.[15-19].x` and `16.[10-12].x`. Current supported versions of Node.js are `14.20.x`, `16.13.x` and `18.10.x`.

PR Close #47730

bob-watson reviewed (changes requested):
LGTM.

Please squash the commits into a single commit with a commit messages that starts with `docs: `

Thanks!

devversion committed:
build: wire up new github action for identifying g3-affecting PRs (#47735)

Wire up new github action for identifying g3-affecting PRs.

PR Close #47735

atcastle committed:
feat(common): Add fill mode to NgOptimizedImage (#47738)

Add a new boolean attribute to NgOptimizedImage called `fill` which does the following:
* Removes the requirement for height and width
* Adds inline styling to cause the image to fill its containing element
* Adds a default `sizes` value of `100vw` which will cause the image to have a responsive srcset automatically generated

PR Close #47738

alan-agius4 committed:
fix(compiler-cli): add missing period to error message (#47744)

With this change we add a missing period to the error message.

PR Close #47744

josephperrott committed:
build: update labels to match standardized labels (#47747)

Update labels as part of standardization

PR Close #47747

josephperrott committed:
build: update to the latest version of the feature triage action

Update to the latest version of the feature triage action

josephperrott committed:
build: update comp labels to be area instead (#47750)

Updates all of the comp: * labels to area: * instead

PR Close #47750

thePunderWoman committed:
docs: release notes for the v14.2.6 release

thePunderWoman committed:
release: bump the next branch to v15.1.0-next.0

thePunderWoman committed:
docs: release notes for the v15.0.0-next.6 release

AndrewKushnir committed:
refactor(common): make `provideHttpClient` return `EnvironmentProviders` (#47758)

This commit updates the `provideHttpClient` function to return the `EnvironmentProviders` instead of a regular `Provider[]`, to make sure that the `provideHttpClient` can only be used where an environment is being setup.

PR Close #47758

josephperrott committed:
build: update commit message based labels action

Update to the latest version of the action

regulas1 committed:
Merge branch 'angular:main' into DocumentationUpdate47179

regulas1 committed:
Merge branch 'DocumentationUpdate47179' of https://github.com/regulas1/angular into DocumentationUpdate47179

jessicajaniuk commented:
@regulas1 You will have to rebase and fix your commit history on this PR. You've merged changes from upstream/main rather than rebasing, and that's brought in a ton of extra commits. Please either rebase and squash or close and recreate a new clean PR.

regulas1 commented:
Sorry about that trying to fix it, Ill create  a new PR
